src/stdlib_linalg_cholesky.fypp

```diff
@@ -0,0 +1,178 @@
+             ! Compute factorization
+             call potrf(triangle,n,a,lda,info)
+             call handle_potrf_info(info,triangle,lda,n,err0)
+                      
+             ! Zero-out the unused part of matrix A
+             clean_unused: if (other_zeroed_ .and. err0%ok()) then    
+                 if (lower_) then 
+                    forall (j=2:n) a(:j-1,j) = zero
+                 else
+                    forall (j=1:n-1) a(j+1:,j) = zero
+                 endif
+             endif clean_unused  
+         
+         endif 
+
+         ! Process output and return
+         call linalg_error_handling(err0,err)
+
+     end subroutine stdlib_linalg_${ri}$_cholesky_inplace
+
+     ! Compute the Cholesky factorization of a symmetric / Hermitian matrix, A = L*L^T = U^T*U. 
+     ! The factorization is returned as a separate matrix
+     pure module subroutine stdlib_linalg_${ri}$_cholesky(a,c,lower,other_zeroed,err) 
+         !> Input matrix a[n,n]
+         ${rt}$, intent(in) :: a(:,:)
+         !> Output matrix with Cholesky factors c[n,n]
+         ${rt}$, intent(out) :: c(:,:)         
+         !> [optional] is the lower or upper triangular factor required? Default = lower
+         logical(lk), optional, intent(in) :: lower
+         !> [optional] should the unused half of the return matrix be zeroed out? Default: yes
+         logical(lk), optional, intent(in) :: other_zeroed
+         !> [optional] state return flag. On error if not requested, the code will stop
+         type(linalg_state_type), optional, intent(out) :: err
+         
+         type(linalg_state_type) :: err0
+         integer(ilp) :: lda,n,ldc,nc
+         
+         ! Check C sizes
+         lda = size(a,1,kind=ilp)
+           n = size(a,2,kind=ilp)
+         ldc = size(c,1,kind=ilp)
+          nc = size(c,2,kind=ilp)
+          
+         if (lda<1 .or. n<1 .or. lda<n .or. ldc<n .or. nc<n) then 
+            
+            err0 = linalg_state_type(this,LINALG_VALUE_ERROR, &
+                                     'invalid matrix sizes: a=',[lda,n], &
+                                                          ' c=',[ldc,nc])
+        
+         else
+            
+            ! Copy data in
+            c(:n,:n) = a(:n,:n)
+            
+            ! Get cholesky factors
+            call stdlib_linalg_${ri}$_cholesky_inplace(c,lower,other_zeroed,err0)            
+            
+         end if            
+         
+         ! Process output and return
+         call linalg_error_handling(err0,err)
+         
+     end subroutine stdlib_linalg_${ri}$_cholesky
+
+     ! Compute the Cholesky factorization of a symmetric / Hermitian matrix, A = L*L^T = U^T*U. 
+     ! Function interface
+     pure module function stdlib_linalg_${ri}$_cholesky_fun(a,lower,other_zeroed) result(c)
+         !> Input matrix a[n,n]
+         ${rt}$, intent(in) :: a(:,:)
+         !> Output matrix with Cholesky factors c[n,n]
+         ${rt}$, allocatable :: c(:,:)         
```

Comment:
```suggestion
         ${rt}$ :: c(size(a, 1),size(a, 2))         
```
This strategy is used in other modules of stdlib. This should avoid the user to know that an allocatable array is used as output